app/scripts/controllers/detect-tokens.js

```diff
@@ -79,7 +80,13 @@ export default class DetectTokensController {
     }
 
     const { tokenList } = this._tokenList.state;
-    if (Object.keys(tokenList).length === 0) {
+    // since the token detection is currently enabled only on Mainnet
+    // we can use the chainId check to ensure token detection is not triggered for any other network
+    // but once the balance check contract for other networks are deploayed and ready to use, we need to update this check.
+    if (
+      this._network.store.getState().provider.chainId !== MAINNET_CHAIN_ID ||
```

Comment:
looks correct (note to other readers that this is different from the above suggestion)

Reply:
this is what I had in mind!